akka-persistence-testkit/src/main/scala/akka/persistence/testkit/internal/TestKitStorage.scala

```diff
@@ -35,8 +36,8 @@ sealed trait InMemStorage[K, R] extends InternalReprSupport[R] {
 
   import scala.math._
 
-  private final val eventsMap: ConcurrentHashMap[K, (Long, Vector[InternalRepr])] =
-    new ConcurrentHashMap()
+  private final val expectNextQueue: mutable.Map[K, util.Queue[InternalRepr]] = mutable.HashMap.empty
+  private final val eventsMap: mutable.Map[K, (Long, Vector[InternalRepr])] = mutable.HashMap.empty
```

Comment:
since we are aiming for simplicity and not performance we should use ordinary immutable Scala collections

```
import scala.collection.immutable.Queue

private var expectNextQueue: Map[K, Queue[InternalRepr]] = Map.empty
private var eventsMap: Map[K, (Long, Vector[InternalRepr])] = Map.empty
```